qemu/tests/single_driver_install.py

```diff
@@ -0,0 +1,148 @@
+import re
+import logging
+from virttest import utils_misc
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    This Test is mainly used as subtests
+    1) Boot up VM
+    2) Uninstall driver (Optional)
+    3) Reboot vm (Based on step 2)
+    4) Update / Downgrade / Install driver
+    5) Reboot vm
+    6) Backup driver installation logs
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def show_file_extentions(session):
+        """
+        Show file extentions in windows guest.
+
+        :param session: VM session.
+        """
+        show_file_ext_cmd = params.get("show_file_ext_cmd")
+        session.cmd(show_file_ext_cmd)
+
+    def get_installation_logs(session):
+        """
+        Get driver installation log.
+
+        :param session: VM session.
+        """
+        tmp_debug = r"C:\Windows\DPINST.log, C:\driver_install.log"
+        driver_logs = params.get("driver_debug_file", tmp_debug).split(',')
+        for _file in driver_logs:
+            status = session.cmd_status("dir %s" % _file)
+            if not status:
+                output = session.cmd("type %s" % _file)
+                _file_host = re.split(r'\\', '%r' % _file)[-1]
+                _debug_log = open("%s/%s" % (test.resultsdir, _file_host), "w")
+                _debug_log.write("%s\n" % output)
+                _debug_log.close()
+
+    def reboot(vm, session=None):
+        """
+        Reboot guest.
+
+        :param vm: VM onject.
+        :param session: VM session.
+        """
+        nic_idx = len(vm.virtnet) - 1
+        while nic_idx >= 0:
+            try:
+                return vm.reboot(session, nic_index=nic_idx)
+            except Exception:
```

Comment:
Thanks @xutian 

reboot() works well without param session, and I have updated the code.
Could you help review it again?

13:35:12 INFO | PASS smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u1.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.single_driver_install.virtio_install.with_netkvm


Thanks.